libs/@blockprotocol/service/src/scripts/generate-service-module-json.ts

```diff
@@ -0,0 +1,232 @@
+import { writeFileSync } from "node:fs";
+/* eslint-disable import/no-extraneous-dependencies */
+import { resolve } from "node:path";
+
+import { generateSchema, getProgramFromFiles } from "typescript-json-schema";
+
+export type ServiceJsonMessage = {
+  messageName: string;
+  description: string;
+  source: "block" | "embedder";
+  respondedToBy: string | null;
+  sentOnInitialization: boolean;
+  data: any;
+  errorCodes: string[];
+};
+
+const program = getProgramFromFiles([resolve("./src/types.ts")]);
+
+const generateServiceSchema = (params: { typeName: string }) => {
+  const schema = generateSchema(program, params.typeName, {
+    required: true,
+    ref: false,
+  });
+
+  if (!schema) {
+    throw new Error(
+      `The JSON Schema for the "${params.typeName}" TS type could not be generated.`,
+    );
+  }
+
+  return schema;
+};
+
+export const mapboxMessages: ServiceJsonMessage[] = [
+  /** Mapbox Geocoding API */
+  {
+    messageName: "mapboxForwardGeocoding",
+    description: "Forward geocoding request",
+    source: "block",
+    respondedToBy: "mapboxForwardGeocodingResponse",
+    sentOnInitialization: false,
+    data: generateServiceSchema({
+      typeName: "MapboxForwardGeocodingData",
+    }),
+    errorCodes: [],
```

Comment:
I'm guessing we'll add some common set of error codes as part of your error handling followup? e.g. `INVALID_TOKEN`, `RATE_LIMITED`, etc

Reply:
Yes, I've captured it in [this follow-up task](https://app.asana.com/0/0/1204004000163863/f) for now. This may require some discussion regarding whether we want to also account for possible mapbox/openAI errors